packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts

```diff
@@ -372,14 +372,15 @@ export class SelectionSetToObject<Config extends ParsedDocumentsConfig = ParsedD
     for (const { field, selectedFieldType } of linkFieldSelectionSets.values()) {
       const realSelectedFieldType = getBaseType(selectedFieldType as any);
       const selectionSet = this.createNext(realSelectedFieldType, field.selectionSet);
+      const isConditional = hasConditionalDirectives(field.directives);
 
       linkFields.push({
         alias: field.alias ? this._processor.config.formatNamedField(field.alias.value, selectedFieldType) : undefined,
-        name: this._processor.config.formatNamedField(field.name.value, selectedFieldType),
+        name: this._processor.config.formatNamedField(field.name.value, selectedFieldType, isConditional),
         type: realSelectedFieldType.name,
         selectionSet: this._processor.config.wrapTypeWithModifiers(
           selectionSet.transformSelectionSet().split(`\n`).join(`\n  `),
-          selectedFieldType
+          isConditional ? realSelectedFieldType : selectedFieldType
```

Comment:
Breaks arrays as this strips the interable wrapper (Array<T>) from the base type when in a directive. 

EG if people is a list: 
```
people @include(if: $withPeople) { 
 id
}
```
Result is that in typescript the output for People is no longer Maybe<Array<Person>> but just Maybe<Person>